includes/integration.php

```diff
@@ -0,0 +1,543 @@
+            if ($removed_cart_item) {
+                $product_id = $removed_cart_item['data']->get_id();
+                $quantity = (int)$removed_cart_item['quantity'];
+                $this->put_event_in_cookie_queue("window.metrilo.removeFromCart('$product_id', $quantity);");
+            }
+        }
+        
+        public function new_order_event($order_id) {
+            try {
+                // fetch the order
+                $order                    = new WC_Order($order_id);
+                $serialized_order         = $this->order_serializer->serialize($order);
+                $client                   = $this->api_client->get_client();
+                $this->identify_call_data = $order->get_billing_email();
+                
+                $client->order($serialized_order);
+                $this->session_set(
+                    $this->get_do_identify_cookie_name(),
+                    wp_json_encode($this->identify_call_data, JSON_UNESCAPED_UNICODE)
+                );
+            } catch(\Exception $e) {
+                $this->error_logger('new_order_event ', $e->getMessage(), $this->plugin_log_path);
+            }
+        }
+        
+        public function resolve_product($product_id)
+        {
+            if (function_exists('wc_get_product')) {
+                return wc_get_product($product_id);
+            }
+        }
+        
+        public function put_event_in_queue($event)
+        {
+            if ($this->check_if_event_should_be_ignored($event)) {
+                return true;
+            }
+            
+            array_push($this->events_queue, $event);
+        }
+        
+        public function put_event_in_cookie_queue($event)
+        {
+            if ($this->check_if_event_should_be_ignored($event)) {
+                return true;
+            }
+            
+            $this->add_item_to_cookie($event);
+        }
+        
+        public function check_if_event_should_be_ignored($event)
+        {
+            if (empty($this->ignore_for_events)) {
+                return false;
+            }
+            if (in_array($event, $this->ignore_for_events)) {
+                return true;
+            }
+            return false;
+        }
+        
+        /**
+         *
+         *
+         * Session and cookie handling
+         *
+         *
+         */
+        
+        public function session_get($k)
+        {
+            if (!is_object($this->woo->session)) {
+                return isset($_COOKIE[$k]) ? $_COOKIE[$k] : false;
+            }
+            return $this->woo->session->get($k);
+        }
+        
+        public function session_set($k, $v)
+        {
+            if (!is_object($this->woo->session)) {
+                @setcookie($k, $v, time() + 43200, COOKIEPATH, COOKIE_DOMAIN);
+                $_COOKIE[$k] = $v;
+                return true;
+            }
+            return $this->woo->session->set($k, $v);
+        }
```

Comment:
```suggestion
        public function session_set($k, $v)
        {
            if (!is_object($this->woo->session)) {
                @setcookie($k, $v, time() + 43200, COOKIEPATH, COOKIE_DOMAIN);
                $_COOKIE[$k] = $v;
            } else {
                $this->woo->session->set($k, $v);
            }
        }
```
The same can be applied in other methods as well. LIke the `session_get` above.

Reply:
@nedick007 what about [this suggestion](https://github.com/Metrilo/metrilo-woocommerce-integration/pull/49/files#r556540315)? ^